File: rule-ai-system/backend/app/models/rule.py
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field

class RuleCondition(BaseModel):
    """룰 조건 모델"""
    field: str
    operator: str
    value: Any
    conditions: Optional[List['RuleCondition']] = None
    parent_operator: Optional[str] = None

class RuleAction(BaseModel):
    """Rule action model"""
    action_type: str = Field(..., description="Type of action to perform")
    parameters: Dict[str, Any] = Field(default_factory=dict, description="Action parameters")

class Rule(BaseModel):
    """룰 모델"""
    name: str
    description: Optional[str] = None
    conditions: List[RuleCondition]
    action: Optional[Dict[str, Any]] = None
    id: Optional[str] = Field(None, description="Rule unique identifier")
    priority: int = Field(default=1, description="Rule execution priority (lower means higher priority)")
    enabled: bool = Field(default=True, description="Whether the rule is enabled or not")
    
    class Config:
        from_attributes = True

# 순환 참조 해결
RuleCondition.model_rebuild() 

File: rule-ai-system/frontend/src/types/rule.ts
export interface RuleCondition {
  field: string;
  operator: string;
  value: any;
}

export interface RuleAction {
  action_type: string;
  parameters: Record<string, any>;
}

export interface Rule {
  id?: string;
  name: string;
  description: string;
  conditions: RuleCondition[];
  actions: RuleAction[];
  priority: number;
  enabled: boolean;
}

export interface ValidationIssue {
  severity: 'error' | 'warning' | 'info';
  message?: string;
  explanation?: string;
  location?: string;
  suggestion?: string;
  field?: string;
  issue_type?: string;
}

export interface ValidationResult {
  is_valid: boolean;
  summary: string;
  issues: ValidationIssue[];
  issue_counts?: Record<string, number>;
  structure?: {
    depth: number;
    condition_count: number;
    unique_fields: string[];
  };
  complexity_score?: number;
  rule_summary?: string;
}

export interface RuleValidationResponse {
  validation_result: ValidationResult;
  rule_summary: string;
}

export interface RuleReportRequest {
  rule_json: Record<string, any>;
  include_markdown?: boolean;
}

export interface RuleReportResponse {
  report: string;
  rule_id?: string;
  rule_name?: string;
} 